examples/src/main/java/org/bitcoinj/examples/ForwardingService.java

```diff
@@ -168,20 +136,24 @@ private void coinForwardingListener(Wallet wallet, Transaction incomingTx, Coin
      * @param forwardingAddress the address to send to
      * @return A future for a TransactionBroadcast object that completes when relay is acknowledged by peers
      */
-    private CompletableFuture<TransactionBroadcast> forward(Wallet wallet, Transaction incomingTx, Address forwardingAddress) {
+    CompletableFuture<TransactionBroadcast> forward(Wallet wallet, Transaction incomingTx, Address forwardingAddress) {
         // Send coins received in incomingTx onwards by sending exactly the outputs that have been sent to us
         SendRequest sendRequest = SendRequest.emptyWallet(forwardingAddress);
         sendRequest.coinSelector = forwardingCoinSelector(incomingTx.getTxId());
         System.out.printf("Creating outgoing transaction for %s...\n", forwardingAddress);
         return wallet.sendTransaction(sendRequest)
                 .thenCompose(broadcast -> {
-                    System.out.printf("Transaction %s is signed and is being delivered to %s...\n", broadcast.transaction().getTxId(), network);
+                    System.out.printf("Transaction %s is signed and is being delivered to %s...\n", broadcast.transaction().getTxId(), wallet.network());
                     return broadcast.awaitRelayed(); // Wait until peers report they have seen the transaction
```

Comment:
The awaitRelay() I noticed takes very long. I tried looking into the code to see how the number of peers to report transaction is decided. So, if I have 4 required number of node connections, transaction will be sent to 2 and then we wait for the other 2 to report they have also seen the transaction. I guess our lack of control of who gets what when and who we are connected is what makes this process long wait